grails-app/services/com/unifina/service/UserService.groovy

```diff
@@ -96,27 +91,14 @@ class UserService {
 		return user
 	}
 
-	def addRoles(user, List<SecRole> roles = null) {
-		def secConf = grailsApplication.config.grails.plugin.springsecurity
-		ClassLoader cl = this.getClass().getClassLoader()
-
-		def userRoleClass = cl.loadClass(secConf.userLookup.authorityJoinClassName)
-		def roleClass = cl.loadClass(secConf.authority.className)
-
-		if (roles == null) {
-			roles = roleClass.findAllByAuthorityInList(secConf.ui.register.defaultRoleNames)
-			if (roles.size() != secConf.ui.register.defaultRoleNames.size()) {
-				throw new RuntimeException("Roles not found: " + secConf.ui.register.defaultRoleNames)
-			}
-		}
-
-		roles.each { role ->
-			userRoleClass.create user, role
+	def addRoles(SecUser user, List<SecRole> roles = null) {
+		roles.each { SecRole role ->
```

Comment:
**Major finding:**
NPE if `roles == null` (which it is by default). Please do `roles?.each`